src/core/queryObserver.ts

```diff
@@ -148,6 +148,37 @@ export class QueryObserver<
     this.currentQuery.removeObserver(this)
   }
 
+  createQueryResult(
```

Comment:
What if `getCurrentResult` would either return a regular or a tracked version? Then it would be a bit more transparent

Reply:
yes, I think this was my first attempt, but we then access the result of that method (`currentResult`) internally, which would lead to some tracking that we don't want, e.g. here:

or here:

https://github.com/tannerlinsley/react-query/blob/361935a12cec6f36d0bd6ba12e84136c405047c5/src/react/useBaseQuery.ts#L83

https://github.com/tannerlinsley/react-query/blob/361935a12cec6f36d0bd6ba12e84136c405047c5/src/react/useBaseQuery.ts#L89

we really only want to track what the consumer of _useQuery_ uses, so I thought the only safe way is to create the tracked version right at the return of `useBaseQuery`